core/domain/user_services.py

```diff
@@ -67,14 +67,17 @@ class UserSettings(object):
         preferred_language_codes: list(str) or None. Exploration language
             preferences specified by the user.
         preferred_site_language_code: str or None. System language preference.
+        role: str. Role of the user. Role corresponds to actions user can
+            perform.
     """
     def __init__(
             self, user_id, email, username=None, last_agreed_to_terms=None,
             last_started_state_editor_tutorial=None, last_logged_in=None,
             last_created_an_exploration=None,
             last_edited_an_exploration=None, profile_picture_data_url=None,
             user_bio='', subject_interests=None, first_contribution_msec=None,
-            preferred_language_codes=None, preferred_site_language_code=None):
+            preferred_language_codes=None, preferred_site_language_code=None,
+            role='EXPLORATION_EDITOR'):
```

Comment:
I think giving default value to the role is fine. The role has nothing to do with user at time of signup.
On registration, user can get only one role i.e 'EXPLORATION_EDITOR' which can later be changed by super_admin. So, we don't need to take that as an argument anywhere.
The role is mandatory in the sense that it is present for every user but it is not a mandatory argument.